### ISSUE TITLE
Feature request: numeric keypad support

### ISSUE THREAD

**Issue by darktjm** [panda3d/panda3d#1517]
panda3d converts numeric keypad keystrokes to their underlying function (either a number or e.g. a cursor key depending on numlock).  This goes all the way down to the low-level drivers (at least the few I looked at).  Even the raw keystroke is only the underlying keycap (e.g. numpad 4 is raw-4, indistinguishable from the upper row 4 key).  This seems very odd for a supposed game engine.  All keys should be supported.   If it's a different key, it should be distinguishable.  I suggest a (possibly prc-enabled) separate raw key name for numeric keypad keys.  Alternatively, you could provide a virtual modifier flag of some kind.  Either way would probably not break legacy apps.  As is, the best I could do is check for both raw and cooked, and if the raw key is a known numeric keypad key, but the cooked key is different, assume it's a numeric keypad key.  Not only does this not work with keys that have regular equivalents (such as - and enter), but it relies on timing (asking for failure) and only works if numlock is off.  Or I guess I could just patch the drivers I care about to send different key names.

**Comment by rdb** [comment]
I agree. At the very least, the "raw" events should reflect the difference between numpad and regular keys.

**Comment by darktjm** [comment]
I will not be able to log into github any more as of tomorrow, for failure to comply with their 2FA policy, so don't expect any comment or reply from me any more.  I have made a mostly untested patch for myself (Linux) a while ago, but I can't be expected to fix the other drivers, and I don't do fork+pull request unless I absolutely have to.  I expected someone familiar with the code to make a patch.  I am not that person.  I have existing, working generic SDL event code for such things from other projects that I will probably end up using for input processing, anyway.
[keypad-linux.patch.txt](https://github.com/panda3d/panda3d/files/12730861/keypad-linux.patch.txt)
